File: Frontend/src/app/receipts/models/receipt.ts
import { GroceryItem } from '../../groceries/models/grocery-item';

export interface Receipt {
    id: string;
    imagePath: string;
    userId: string;
    items: GroceryItem[];
    createdAt: Date;
    storeName?: string;
    purchaseDate?: Date;
    totalAmount?: number;
}



export interface OcrResult {
  storeName: string;
  purchaseDate: Date;
  totalAmount: number;
  items: GroceryItem[];
  rawText: string;
  confidence: number;
}

File: Backend/internal/models/receipt.go
package models

import (
	"time"
)

type Receipt struct {
	ID           uint          `gorm:"primaryKey" json:"id"`
	UserID       uint          `json:"user_id"`
	ImagePath    string        `gorm:"not null" json:"image_path"`
	TotalAmount  float64       `json:"total_amount"`
	StoreName    string        `json:"store_name"`
	PurchaseDate time.Time     `json:"purchase_date"`
	CreatedAt    time.Time     `json:"created_at"`
	Items        []GroceryItem `gorm:"foreignKey:ReceiptID" json:"items"`
}
